contracts/colonyNetwork/ColonyNetworkDeployer.sol

```diff
@@ -175,6 +178,94 @@ contract ColonyNetworkDeployer is ColonyNetworkStorage {
     // This is intentional, as we want to allow the same Colony to be deployed on different chains
   }
 
+  function setDomainTokenReceiverResolver(address _resolver) public stoppable auth {
+    domainReceiverResolverAddress = _resolver;
+  }
+
+  function getDomainTokenReceiverResolver() public view returns (address) {
+    return domainReceiverResolverAddress;
+  }
+
+  function checkDomainTokenReceiverDeployed(
+    uint256 _domainId
+  ) public stoppable calledByColony returns (address domainTokenReceiverAddress) {
+    // Calculate the address the domain should be receiving funds at
+    domainTokenReceiverAddress = getDomainTokenReceiverAddress(msgSender(), _domainId);
+
+    if (!isContract(domainTokenReceiverAddress)) {
+      // Then deploy the contract
+      bytes32 salt = getDomainTokenReceiverDeploySalt(msgSender(), _domainId);
+      address deployedAddress = ICreateX(CREATEX_ADDRESS).deployCreate3AndInit(
+        salt,
+        type(EtherRouterCreate3).creationCode,
+        abi.encodeWithSignature("setOwner(address)", (address(this))),
+        ICreateX.Values(0, 0)
+      );
+      require(
+        deployedAddress == domainTokenReceiverAddress,
+        "colony-network-domain-receiver-deploy-wrong-address"
+      );
+    }
+
+    // Check it's got the right resolver
+    try EtherRouter(payable(domainTokenReceiverAddress)).resolver() returns (Resolver resolver) {
+      if (address(resolver) != domainReceiverResolverAddress) {
+        EtherRouter(payable(domainTokenReceiverAddress)).setResolver(domainReceiverResolverAddress);
+      }
+    } catch {
+      revert("colony-network-domain-receiver-not-etherrouter");
+    }
+
+    // Check it's set up correctly
```

Comment:
I'd feel comfortable rolling this up into the deploy block earlier. Realistically will this value ever be changed?

Reply:
It shouldn't be, but arbitrary transactions can do arbitrary things. In general, however, I've been taking a bit more of a belt-and-braces approach when contracts at deterministic addresses are involved, hence leaving it as a separate block.